### ISSUE TITLE
`expose_split` exposures different than `expose_cy` when truncated

### ISSUE THREAD

**Issue by MatthewCaseres** [mattheaphy/actxps#46]
```r
df1 <- toy_census %>% filter(pol_num == 1) %>% expose_cy(
  start_date = "2007-6-15",
  end_date = "2019-02-27",
  target_status = "Surrender") %>% 
  select(pol_num, cal_yr, cal_yr_end, term_date, issue_date, exposure, status)

df2 <-df1 %>% 
  expose_split() %>%
  select(pol_num, cal_yr, cal_yr_end, issue_date, exposure_cal, status)
```

output data

```
> df1 %>% tail(2)
Exposure data

 Exposure type: calendar_year 
 Target status: Surrender 
 Study range: 2007-06-15 to 2019-02-27

# A tibble: 2 × 7
  pol_num cal_yr     cal_yr_end term_date issue_date exposure status
    <int> <date>     <date>     <date>    <date>        <dbl> <fct> 
1       1 2018-01-01 2018-12-31 NA        2010-01-01    1     Active
2       1 2019-01-01 2019-12-31 NA        2010-01-01    0.159 Active
> df2 %>% tail(2)
Exposure data

 Exposure type: split_year 
 Target status: Surrender 
 Study range: 2007-06-15 to 2019-02-27

# A tibble: 2 × 6
  pol_num cal_yr     cal_yr_end issue_date exposure_cal status
    <int> <date>     <date>     <date>            <dbl> <fct> 
1       1 2018-01-01 2018-12-31 2010-01-01            1 Active
2       1 2019-01-01 2019-12-31 2010-01-01            1 Active
```

pol_num 2 has similar issues, issue is not specific to issue_date on new years

**Comment by MatthewCaseres** [comment]
Something similar happens with start dates as well:

```r
df1 <- toy_census %>% filter(pol_num == 1) %>% expose_cy(
  start_date = "2011-11-1",
  end_date = "2013-02-27",
  target_status = "Surrender") %>% 
  select(pol_num, cal_yr, cal_yr_end, term_date, issue_date, exposure, status)

df2 <-df1 %>%
  expose_split() %>%
  select(pol_num, cal_yr, cal_yr_end, issue_date, exposure_cal, status)
```
gives
```
> df1
  pol_num cal_yr     cal_yr_end term_date issue_date exposure status
    <int> <date>     <date>     <date>    <date>        <dbl> <fct> 
1       1 2011-01-01 2011-12-31 NA        2010-01-01    0.167 Active
2       1 2012-01-01 2012-12-31 NA        2010-01-01    1     Active
3       1 2013-01-01 2013-12-31 NA        2010-01-01    0.159 Active
>df2
  pol_num cal_yr     cal_yr_end issue_date exposure_cal status
    <int> <date>     <date>     <date>            <dbl> <fct> 
1       1 2011-01-01 2011-12-31 2010-01-01            1 Active
2       1 2012-01-01 2012-12-31 2010-01-01            1 Active
3       1 2013-01-01 2013-12-31 2010-01-01            1 Active
```

**Comment by mattheaphy** [comment]
Thanks for pointing this out. It's going to be patched in release 1.5.0. Both your examples above are now working as intended in that version. I also added new tests to the package to ensure that start / end dates are respected.

## Example #1

```
> df2
Exposure data

 Exposure type: split_year 
 Target status: Surrender 
 Study range: 2007-06-15 to 2019-02-27

# A tibble: 2 × 6
  pol_num cal_yr     cal_yr_end issue_date exposure_cal status
    <int> <date>     <date>     <date>            <dbl> <fct> 
1       1 2018-01-01 2018-12-31 2010-01-01        1     Active
2       1 2019-01-01 2019-02-27 2010-01-01        0.159 Active
```

## Example #2

```
> df2
Exposure data

 Exposure type: split_year 
 Target status: Surrender 
 Study range: 2011-11-01 to 2013-02-27

# A tibble: 3 × 6
  pol_num cal_yr     cal_yr_end issue_date exposure_cal status
    <int> <date>     <date>     <date>            <dbl> <fct> 
1       1 2011-11-01 2011-12-31 2010-01-01        0.167 Active
2       1 2012-01-01 2012-12-31 2010-01-01        1     Active
3       1 2013-01-01 2013-02-27 2010-01-01        0.159 Active
```
